classes/output/renderer.php

```diff
@@ -0,0 +1,34 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Renderer class for logstore_xapi
+ *
+ * @package    logstore_xapi
+ * @copyright  2020 Learning Pool Ltd <https://learningpool.com/>
+ * @author     Stephen O'Hara <stephen.ohara@learningpool.com>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace logstore_xapi\output;
+
+defined('MOODLE_INTERNAL') || die();
+
+use plugin_renderer_base;
+
+class renderer extends plugin_renderer_base {
```

Comment:
the name should be logstore_xapi_renderer

Reply:
I'm using namespace, see line 26

Reply:
https://docs.moodle.org/dev/Overriding_a_renderer

> There is also a routine way to name renderers much like the process for naming a theme's overriding renderer.
> 
> It should always be made up of the following three components joined by underscores:
> 
> component This is the component the renderer belongs too, e.g. block, mod, or core.
> name This is the name of what ever the component is, e.g. forum, workshop, or wiki.
> renderer They should always be completed by the word renderer.

Reply:
https://docs.moodle.org/dev/Overriding_a_renderer#Namespaces

Reply:
Yes, thats true. 

Renderers can also be used with code that uses namespaces.

Reply:
So you need to use namespace and follow the naming convention.